lib/model/content.dart

```diff
@@ -576,27 +578,64 @@ class ImagePreviewNode extends BlockContentNode {
   bool operator ==(Object other) {
     return other is ImagePreviewNode
       && other.srcUrl == srcUrl
-      && other.thumbnailUrl == thumbnailUrl
+      && other.thumbnail == thumbnail
       && other.loading == loading
       && other.originalWidth == originalWidth
       && other.originalHeight == originalHeight;
   }
 
   @override
   int get hashCode => Object.hash('ImagePreviewNode',
-    srcUrl, thumbnailUrl, loading, originalWidth, originalHeight);
+    srcUrl, thumbnail, loading, originalWidth, originalHeight);
 
   @override
   void debugFillProperties(DiagnosticPropertiesBuilder properties) {
     super.debugFillProperties(properties);
     properties.add(StringProperty('srcUrl', srcUrl));
-    properties.add(StringProperty('thumbnailUrl', thumbnailUrl));
+    properties.add(DiagnosticsProperty<ImageThumbnailLocator>('thumbnail', thumbnail));
     properties.add(FlagProperty('loading', value: loading, ifTrue: "is loading"));
     properties.add(DoubleProperty('originalWidth', originalWidth));
     properties.add(DoubleProperty('originalHeight', originalHeight));
   }
 }
 
+/// Data to locate an image thumbnail,
+/// and whether the image has an animated version.
+///
+/// Use [ImageThumbnailLocatorExtension.resolve] to obtain a suitable URL
+/// for the current UI need.
+@immutable
+class ImageThumbnailLocator extends DiagnosticableTree {
+  ImageThumbnailLocator({
+    required this.urlPath,
+    required this.hasAnimatedVersion,
+  }) : assert(urlPath.startsWith(urlPathPrefix));
+
+  final String urlPath;
+  final bool hasAnimatedVersion;
+
+  static const urlPathPrefix = '/user_uploads/thumbnail/';
+
+  @override
+  bool operator ==(Object other) {
+    if (other is! ImageThumbnailLocator) return false;
+    return urlPath == other.urlPath
+      && hasAnimatedVersion == other.hasAnimatedVersion;
+  }
+
+  @override
+  int get hashCode => Object.hash(urlPath, hasAnimatedVersion);
```

Comment:
In hashCode, seed the hash with the class name so that the hash doesn't collide with other objects that happen to also consist of a string and a bool:
```suggestion
  int get hashCode => Object.hash('ImageThumbnailLocator', urlPath, hasAnimatedVersion);
```